appjs/views/edit_project.js

```diff
@@ -404,16 +483,19 @@ var EditProject = BaseView.extend(require('./mixins/actions'), require('./mixins
         if ( view.model.hasType( 'graphic' ) || view.model.hasPreviewType('live') ) {
           // Setup our iframe with pym
           if ( view.pym ) { view.pym.remove(); }
-          view.pym = new pym.Parent('embed-preview', previewUrl);
-          view.pym.iframe.onload = iframeLoaded;
-
+          if ( view.formValidate(view.model, view.$('#projectForm')) ){
+            view.pym = new pym.Parent('embed-preview', previewUrl);
+            view.pym.iframe.onload = iframeLoaded;
+          }
           // In case some dumb script hangs the loading process
           setTimeout(iframeLoaded, 20000);
-
-          if ( view.togglePreview ){
-            $( "#draft-preview" ).trigger( "click" );
+        }
+        if(view.model.hasPreviewType('live')){
+          if(view.model.getConfig().spreadsheet_template){
+            $( "input[name='google_doc_url']" ).after('<button type="button" id="spreadsheet-button" data-hook="create-spreadsheet" class="btn btn-default">Get new spreadsheet</button>');
           }
         }
+        view.getTwitterCount();
```

Comment:
Get the twitter count for `#shareText` on project render so that it appears automatically and doesn't require the user to type before showing the remaining character count.